### PR TITLE
Implement a memcpy override function that havocs one byte of the dst array

### PR DESCRIPTION
**Pull request by angelhof**
Implement a `memcpy` stub that is meant to replace `memcpy_override_no_op`. It assigns a nondeterministic byte to a nondeterministic index of the destination array so that assertions on the values of the `memcpy` destination array fail (as they should).

Replaced `memcpy_override_no_op` with the new version is several harnesses and the running time increases only slightly. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .cbmc-batch/stubs/memcpy_override_havoc.c [hunk 1]
```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (danielsn) [inline .cbmc-batch/stubs/memcpy_override_havoc.c]: 2019

File: .cbmc-batch/stubs/memcpy_override_havoc.c [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    __CPROVER_precondition(
+        __CPROVER_POINTER_OBJECT(dst) != __CPROVER_POINTER_OBJECT(src) ||
+            ((const char *)src >= (const char *)dst + n) || ((const char *)dst >= (const char *)src + n),
+        "memcpy src/dst overlap");
+    __CPROVER_precondition(src != NULL && __CPROVER_r_ok(src, n), "memcpy source region readable");
+    __CPROVER_precondition(dst != NULL && __CPROVER_w_ok(dst, n), "memcpy destination region writeable");
+
+    size_t index;
+    __CPROVER_assume(index < n);
+    ((uint8_t *)dst)[index] = nondet_uint8_t();
```
Comment (danielsn) [inline .cbmc-batch/stubs/memcpy_override_havoc.c]: what happens if memcpy was called with n == 0?

Comment (angelhof) [reply .cbmc-batch/stubs/memcpy_override_havoc.c]: You are right. If `n == 0` there should be no "havoc".

**Comment by angelhof** [comment]
Could you review this @tautschnig @karkhaz ?

**Comment by tautschnig** [comment]
@angelhof Could you please rebase to bring it onto current master and thereby also re-trigger CI?

**Pull request merged**